src/viam/resource/types.py

```diff
@@ -196,4 +196,5 @@ def resource_name_from_string(string: str) -> ResourceName:
     return ResourceName(namespace=parts[0], type=parts[1], subtype=parts[2], name=name)
 
 
+Validator: TypeAlias = Callable
```

Comment:
I wonder if we can be any more strict with this type definition. Perhaps `Callable[[ComponentConfig], List[str]]`?

Reply:
I agree. I was doing that at first and it was giving me some Type errors so I kept it that way just for now.

Reply:
Bumping this + wondering if we can do a more strict type; no worries if not.

Reply:
We should definitely make this more strict. And again, as above, this should return a Sequence rather than a List

I also think it should be below `ResourceCreator` for alphabetical reasons :)